plugins/amqp/amqp_input.go

```diff
@@ -203,7 +203,8 @@ func (ai *AMQPInput) Run(ir InputRunner, h PluginHelper) (err error) {
 		}
 		msg.Ack(false)
 	}
-	return nil
+	// return an error message to trigger a potential restart of the plugin
+	return fmt.Errorf("Channel closed while reading from queue %s",  ai.config.Queue)
```

Comment:
Wouldn't this return an error even during normal shutdown? We need a way to distinguish btn the AMQP channel closing unexpectedly and it closing as expected b/c the `Stop` method was called.